System/RTETools.cpp

```diff
@@ -107,4 +107,28 @@ namespace RTE {
 	bool WithinBox(Vector &point, Vector &boxPos, float width, float height) {
 		return point.m_X >= boxPos.m_X && point.m_X < (boxPos.m_X + width) && point.m_Y >= boxPos.m_Y && point.m_Y < (boxPos.m_Y + height);
 	}
-}
\ No newline at end of file
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	std::string RoundFloatToPrecision(float input, int precision, int roundingMode) {
+		if (roundingMode == 0) {
+			std::stringstream floatStream;
+			floatStream << std::fixed << std::setprecision(precision) << input;
+			return floatStream.str();
+		} else {
+			RTEAssert(powf(10.0F, static_cast<float>(precision)) < std::numeric_limits<float>::max(), "PRECISION SET GREATER THAN EXPONENT MAX!");
```

Comment:
lol no need to capitalize all these asserts
Also, I don't really like having all these asserts redoing similar calculations. Extract the `powf(10.0F, static_cast<float>(precision))` into a variable and reuse it.